metrictank/src/main/java/com/expedia/metrics/metrictank/MetricPointKafkaDeserializer.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.metrics.metrictank;
+
+import java.util.Arrays;
+import java.util.Map;
+
+public class MetricPointKafkaDeserializer extends AbstractMetricTankDeserializer<MetricPoint> {
+
+    @Override
+    public void configure(Map configs, boolean isKey) {
+
+    }
+
+    @Override
+    public MetricPoint deserialize(String topic, byte[] data) {
+        try {
+            //Refer to
+            // 1 - https://github.com/raintank/schema/blob/master/msg/msg.go#L114
+            // 2 - https://github.com/raintank/schema/blob/master/msg/format.go
+            Format format = Format.values()[data[0]];
+            if (data.length == V2_TYPE_SIZE && format == Format.FormatMetricPoint) {
+                return deserializeV2(Arrays.copyOfRange(data, 1, data.length));
+            }
+        }
+        catch (Exception e) {
+            return new MetricPoint(new MetricKey(0, e.getMessage()), 0d, 0l);
```

Comment:
This hides the Exception and will lead to bugs.

The caller should decide how to handle the Exception so don't catch it (or catch it and wrap in an IOException).

Throwing an IOException means this can't implement Kafka `Deserializer`. I'm OK with that, make this a general purpose deserializer and let the caller wrap it in a `Deserializer` and make their own decision about what to do with Exceptions. Bonus: Then you can get rid of the dependency on kafka.